src/Purebred/System/Process.hs

```diff
@@ -82,19 +89,44 @@ readProcess
   -> m (ExitCode, Tainted LB.ByteString, Tainted LB.ByteString)
 readProcess = (fmap . fmap) (bimap taint taint) System.Process.Typed.readProcess
 
+runEntityCommand ::
+     (MonadError Error m, MonadIO m)
+  => EntityCommand m a
+  -> AppState
+  -> m AppState
+runEntityCommand cmd s =
+  let acquire = view (ccResource . rsAcquire) cmd
+      update = view (ccResource . rsUpdate) cmd
+      free = view (ccResource . rsFree) cmd
+   in do tmpfile <- acquire
+         update tmpfile (view ccEntity cmd)
+         s' <-
+           tryReadProcess (view ccProcessConfig cmd (view ccEntity cmd) tmpfile) >>=
+           (flip (view ccAfterExit cmd) tmpfile <$> handleExitCode s)
+         free tmpfile
```

Comment:
Hm... now looking at this I wonder if it's doing what I want it to do ...

Reply:
I do wonder if we shouldn't use `bracket` (or similar) here, so that resource can be properly cleaned up if exception occur during `update` or `tryReadProcess`.

Reply:
right? I also looked at it and thought: "Hm... I don't think the `free` is called here." Haven't had a chance to verify it tho.

Will check. Cheers for the pointer. Already forgot about it.

Reply:
Used https://hackage.haskell.org/package/exceptions-0.10.2/docs/Control-Monad-Catch.html#v:bracket since the `bracket` from `Control.Exception` is `IO` only.